djvm/src/main/kotlin/net/corda/djvm/rules/implementation/RewriteClassMethods.kt

```diff
@@ -47,20 +47,25 @@ object RewriteClassMethods : Emitter {
                     preventDefault()
                 }
 
-                INVOKESTATIC -> if (isClassForName(instruction)) {
-                    invokeStatic(
-                        owner = "sandbox/java/lang/DJVM",
-                        name = "classForName",
-                        descriptor = instruction.descriptor
-                    )
-                    preventDefault()
+                INVOKESTATIC -> if (instruction.memberName == "forName") {
+                    if (instruction.descriptor == "(Ljava/lang/String;ZLjava/lang/ClassLoader;)Ljava/lang/Class;") {
+                        invokeStatic(
+                            owner = "sandbox/java/lang/DJVM",
+                            name = "classForName",
+                            descriptor = instruction.descriptor
+                        )
+                        preventDefault()
+                    } else if (instruction.descriptor == "(Ljava/lang/String;)Ljava/lang/Class;") {
+                        // Map the class name into the sandbox namespace, but still invoke
+                        // Class.forName(String) here so that it uses the caller's classloader.
+                        invokeStatic(
+                            owner = "sandbox/java/lang/DJVM",
```

Comment:
Doesnt invoking `Class.forName` potentially introduce a gap for loading unsandboxed code?

Reply:
Potentially yes, which is why the DJVM is careful to prepend the class name with `sandbox.` first. Unfortunately we can't forbid code from invoking `Class.forName(...)` and `newInstance()` because this would kill Kotlin.

Reply:
I see, but can the user still define a custom classloader bypassing the DJVM altogether?

Reply:
There already is a comment:
```kotlin
// Map the class name into the sandbox namespace, but still invoke
// Class.forName(String) here so that it uses the caller's classloader.
```

Reply:
Yes but that comment explain what the code does not why it is necessary, as you explained in this PR comments

Reply:
Fixed.